js/vendor/owncloud/share.js

```diff
@@ -145,7 +145,7 @@
 			var dropDownEl;
 			var self = this;
 			var html = '<div id="dropdown" class="drop shareDropDown" data-item-type="' + itemType +
-				'" data-item-source="' + path + '">';
+				'" data-item-source="' + path + '" data-id="' + data[0].file_source + '">';
```

Comment:
@oparoz Do you find any other mistake in the rest of the code? This will be fixed, once your PR arrives.